src/lib/src/ng-select.component.ts

```diff
@@ -317,6 +312,25 @@ export class NgSelectComponent implements OnInit, ControlValueAccessor {
         this.onBlur.emit($event);
     }
 
+    private validateWriteValue(value: any) {
+        if (value instanceof Object && this.valueKey) {
+            throw new Error('Binding object with valueKey is not allowed.')
+        }
+    }
+
+    private selectWriteValue(value: any) {
+        this.validateWriteValue(value);
+        let index = -1;
+        if (this.valueKey) {
+            index = this.itemsList.items.findIndex(x => x[this.valueKey] === value);
+        } else {
+            index = this.itemsList.items.indexOf(value);
+            index = index > -1 ? index :
+                this.itemsList.items.findIndex(x => x[this.labelKey] === value[this.labelKey])
+        }
+        this.itemsList.select(this.itemsList.items[index]);
```

Comment:
There is case then index will still be -1 (autocomplete) and calling `this.itemsList.select` will cause error. Maybe call it only if index > -1